src/utils/opt_in_checker.py

```diff
@@ -0,0 +1,251 @@
+        :return: opt-in dialog result.
+        """
+        start_time = time.time()
+        answer = self._ask_opt_in(self.opt_in_question, self.dialog_timeout)
+        time_passed = time.time() - start_time
+        while time_passed < self.dialog_timeout and answer == DialogResult.TIMEOUT_REACHED:
+            answer = self._ask_opt_in(self.opt_in_question_incorrect_input, self.dialog_timeout - time_passed)
+            time_passed = time.time() - start_time
+
+        if answer == DialogResult.TIMEOUT_REACHED:
+            colored_print(OptInChecker.response_confirmation_timer_reached)
+
+        return answer
+
+    @staticmethod
+    def isip_file_base_dir():
+        """
+        Returns the base directory of the ISIP file.
+        :return: base directory of the ISIP file.
+        """
+        platform = system()
+
+        dir_to_check = None
+
+        if platform == 'Windows':
+            dir_to_check = '$LOCALAPPDATA'
+        elif platform in ['Linux', 'Darwin']:
+            dir_to_check = Path.home()
+
+        if dir_to_check is None:
+            raise Exception('Failed to find location of the ISIP file.')
+
+        return os.path.expandvars(dir_to_check)
+
+    @staticmethod
+    def isip_file_subdirectory():
+        """
+        Returns ISIP file subdirectory.
+        :return: ISIP file subdirectory.
+        """
+        platform = system()
+        if platform == 'Windows':
+            return 'Intel Corporation'
+        elif platform in ['Linux', 'Darwin']:
+            return 'intel'
+        raise Exception('Failed to find location of the ISIP file.')
+
+    def isip_file(self):
+        """
+        Returns the ISIP file path.
+        :return: ISIP file path.
+        """
+        return os.path.join(self.isip_file_base_dir(), self.isip_file_subdirectory(), "isip")
+
+    def create_new_isip_file(self):
+        """
+        Creates a new ISIP file.
+        :return: True if the file is created successfully, otherwise False
+        """
+        if not self.create_or_check_isip_dir():
+            return False
+        try:
+            open(self.isip_file(), 'w').close()
+        except Exception:
+            return False
+        return True
+
+    def create_or_check_isip_dir(self):
+        """
+        Creates ISIP file directory and checks if the directory is writable.
+        :return: True if the directory is created and writable, otherwise False
+        """
+        isip_dir = os.path.join(self.isip_file_base_dir(), self.isip_file_subdirectory())
+        if not os.path.isdir(isip_dir):
+            if not os.access(isip_dir, os.W_OK):
+                print("Failed to update opt-in status. "
+                      "Cannot create a file when that file already exists: {}".format(isip_dir))
+                return
+            os.remove(isip_dir)
+        if not os.path.exists(isip_dir):
+            if not os.access(self.isip_file_base_dir(), os.W_OK):
```

Comment:
base_dir is either home directory on Linux/MacOS or standard "AppData\Local" on Windows, so it is unlikely to be not a directory, but I added this check just in case.